### ISSUE TITLE
Fine Tune Model with fitness score 0.0 and weird folder prefix

### ISSUE THREAD

**Issue by Timmy-Eink** [ultralytics/ultralytics#17456]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Multi-GPU, Hyperparameter Tuning

### Bug

I am using fine tune model with this cmd:

`python -m torch.distributed.run --nproc_per_node 4 /home/timmy/ultralytics/yolo/tune.py --batch 16 --data /home/timmy/ultralytics/data/TM2.yaml --weights /home/timmy/ultralytics/weights/yolo11m-seg.pt --device 0,1,2,3`

I  got folder with prefix -172, I don't know why this happen

```
ls /data/TM2-1111/
-172  -1722  -1723  -1724  -1725  -1726  -1727  -1728  tune  tune2
```

and each iteration I always got 0.0 fitness score

### Environment

```
Ultralytics 8.3.28 🚀 Python-3.10.15 torch-2.4.1 CUDA:0 (NVIDIA L40-16C, 16151MiB)
Setup complete ✅ (16 CPUs, 94.3 GB RAM, 51.1/55.7 GB disk)

OS                  Linux-6.8.0-48-generic-x86_64-with-glibc2.39
Environment         Linux
Python              3.10.15
Install             pip
RAM                 94.29 GB
Disk                51.1/55.7 GB
CPU                 INTEL XEON SILVER 4514Y
CPU count           16
GPU                 NVIDIA L40-16C, 16151MiB
GPU count           4
CUDA                11.8

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.5.2>=3.3.0
opencv-python       ✅ 4.10.0>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.4.1>=1.8.0
torchvision         ✅ 0.19.1>=0.9.0
tqdm                ✅ 4.67.0>=4.64.0
psutil              ✅ 6.1.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ❌ (not installed)>=2.0.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
torch               ✅ 2.4.1!=2.4.0,>=1.8.0; sys_platform == "win32"
```

### Minimal Reproducible Example

tune.py
```

from ultralytics import YOLO
import os
import torch

os.environ['CUDA_LAUNCH_BLOCKING'] = "1"
os.environ['TORCH_USE_CUDA_DSA'] = "1"
os.environ['OMP_NUM_THREADS'] = "1"
os.environ["NCCL_NET"] = "Socket"
os.environ["TORCH_CPP_LOG_LEVEL"] = "1"

torch.cuda.empty_cache()
torch.cuda.empty_cache()

if __name__ == '__main__':
    model = YOLO(model="weights/yolo11m-seg.pt", task="segment")

    model.tune(data="data/TM2.yaml",
                iterations=300,
                epochs=5,
                optimizer="adam",
                project="/data/TM2-1111",
                name="1108-1280",
                imgsz=1280,
                cache=False,
                overlap_mask=False,
                batch=16,
                device=[0,1,2,3],
               )
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Timmy-Eink, thank you for your interest in Ultralytics 🚀! This is an automated response to help guide you. An Ultralytics engineer will also assist with your issue soon.

If you're encountering a 🐛 bug, kindly ensure you've provided a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/). You've shared quite a bit already, but make sure your example captures the exact context of your issue.

For new users, it's helpful to visit the [Docs](https://docs.ultralytics.com) where you can find comprehensive examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which might answer many common questions.

Additionally, if this is about custom training ❓, please ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To stay connected with the Ultralytics community, you're welcome to join us on [Discord](https://ultralytics.com/discord) for real-time discussions 🎧, engage on [Discourse](https://community.ultralytics.com), or check out our [Subreddit](https://reddit.com/r/ultralytics).

## Troubleshooting

Please try upgrading to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Verified Environments

You can also run YOLO in any of the following environments which have all dependencies preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Continuous Integration

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates the current CI status, which tests YOLO across different modes and tasks on macOS, Windows, and Ubuntu every 24 hours and for every commit.

Thank you for your patience and support!

**Comment by Y-T-G** [comment]
> 
> I  got folder with prefix -172, I don't know why this happen
> 
> ```
> ls /data/TM2-1111/
> -172  -1722  -1723  -1724  -1725  -1726  -1727  -1728  tune  tune2
> ```

1108-1280 = -172

Did you pass a string as name?

> 
> and each iteration I always got 0.0 fitness score
> 

What's the actual code you used? You said you used the first code, but in the minimum reproducible example you have a different code.

**Comment by Y-T-G** [comment]
You can try increasing the number of epochs

**Comment by Timmy-Eink** [comment]
Actually I set epoch 5 for testing the folder name, before that my epoch setting is 1000

**Comment by Timmy-Eink** [comment]
I think I found the problem. I set the name parameter, but for some reason, the folder created does not match my specified name, which is causing issues when trying to find it during the fitness calculations.

**Comment by glenn-jocher** [comment]
@Timmy-Eink it seems like the issue might be related to how the folder names are being generated. Ensure that the `name` parameter is correctly set and check for any unintended modifications in the code that might affect folder naming. If the problem persists, consider reviewing the directory creation logic in your script.
